### PR TITLE
Codesign Exokit.app

### LINKED ISSUE #55: MacOS .app bundle is unsigned
Downloading from get.webmr.io yields the following error:

image

### LINKED ISSUE #57: CI: do not codesign PRs
Currently our builds will error during a PR build: https://travis-ci.org/webmixedreality/exokit/builds/377086347?utm_source=slack&utm_medium=notification

This is by design: the PRing Github account does not (and should not) have access to secure variables in the build, such as our code signing keys. But it would be nice for the build to not fail, since build success is highly useful info for PRs.

The straightforward fix for this would be to skip the code signing steps (for [Windows on Appveyor](https://github.com/webmixedreality/exokit/blob/master/appveyor.yml#L39) and [OSX on Travis CI](https://github.com/webmixedreality/exokit/blob/master/.travis.yml#L74) if/when the `PFX_KEY` environment variable is not defined -- and maybe log the fact that this is happening for later debugging in case this behavior leaks into `master`.

CC @shawwn

### PR DESCRIPTION
**Pull request by shawwn**
Closes #55, closes #57.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -63,17 +63,10 @@ matrix:
...
     - rm -R node
     - tar -czf exokit-macos.tar.gz --exclude=".*" --exclude="*.tar.gz" *
-    - security create-keychain -p nopassword build.keychain
-    - security default-keychain -s build.keychain
-    - security unlock-keychain -p nopassword build.keychain
-    - security set-keychain-settings -t 3600 -u build.keychain
-    - security import codesign-macos.pfx -k build.keychain -P "$PFX_KEY" -A
-    - "security set-key-partition-list -S apple-tool:,apple: -s -k nopassword build.keychain"
-    - appdmg /tmp/appdmg.json exokit-macos-x64.dmg
+    - sh codesign-macos.sh
```
Comment (modulesio) [inline .travis.yml]: Can we `./`?

Comment (shawwn) [reply .travis.yml]: I'd prefer not to rely on chmod +x for `codesign-macos.sh`, but it's up to you.

Comment (shawwn) [reply .travis.yml]: Okay, done.

**Comment by shawwn** [comment]
set -x would leak PFX_KEY into the build logs.

**Comment by modulesio** [comment]
@shawwn as far as CI goes, it would be filtered from logs. It’s an automatic Travis feature afair.

Additionally, would it not be possible to devnull it even if that weren’t the case?

**Comment by shawwn** [comment]
My main issue with this setup is that it takes a full hour to sign all the files in .app, and then doesn’t successfully sign it anyway. It complains about node_modules/.bin

The PR might be fine as a way of getting rid of build errors for contributors, but perhaps not super important.

A better approach is probably to zip node_modules somehow. See Zeit/pkg

**Comment by modulesio** [comment]
> A better approach is probably to zip node_modules somehow. See Zeit/pkg

Hm, interesting approach...

**Pull request closed without merging**